### ISSUE TITLE
imbalanced dataset

### ISSUE THREAD

**Issue by Theophylline** [jrzech/reproduce-chexnet#1]
Hi John,

Great work. I am also trying to reproduce the results from the chexnet paper for learning purposes. I just have 2 quick questions:

1. Did you modify the loss function to account for the imbalanced dataset? The original authors used the unweighted version of binary cross entropy, which seems weird to me because negative cases outweigh the positive ones.

2. How long did you train the model for?

Thank you!

**Comment by jrzech** [comment]
Hi Theophylline, thanks -- I'm glad you're finding it useful.

#1, Unweighted binary cross entropy works fine. The dataset is imbalanced,
but binary cross entropy puts an infinite penalty on classifying everything
as negative, and it trains reliably.

#2, Actual training time on the GPU? On a 1080ti, ~4 hours. It will stop
training when validation loss is no longer improving.

**Comment by Theophylline** [comment]
@jrzech Hi John, a quick follow up question: after training densenet-121 for about 20 epochs, it seems that most of the predicted probabilities are quite low, which is likely caused by the imbalanced dataset. Did you have to adjust the thresholds for _each_ of the classes in order to maximize the F1 score? Thanks, I really appreciate your help.

**Comment by jrzech** [comment]
Hi Theophylline -- the generated predictions do indeed tend to be low. I
calculated AUCs, which depend only on the relative ranking of the predicted
probabilities. How you set classification thresholds depends on your goal,
you can certainly pick different thresholds for different targets.
